[OSDOCS#17058]: DC-2: Image Mirroring and oc mirror Tooling-CQA-1 (#115599)
openshift/openshift-docs: disconnected/installing-mirroring-disconnected.adoc

toc::[]

[role="_abstract"]

> You can run your cluster in a restricted network by installing from a mirrored set of {product-title} container images in a private registry. This registry must be running whenever the cluster is running. For more information, see "Prerequisites".

Reviewer: This is a good rewrite but tbh, I don't see why we called out the prereqs section in the first place. So I would just delete this last sentence since users should be looking at that anyways:

```suggestion
You can run your cluster in a restricted network by installing from a mirrored set of {product-title} container images in a private registry. This registry must be running whenever the cluster is running.
```

You can use the oc-mirror {oc-first} plugin to mirror images to a mirror registry in your fully or partially disconnected environments. You must run oc-mirror from a system with internet connectivity in order to download the required images from the official Red Hat registries.

[IMPORTANT]
====

> The oc-mirror v1 plugin is deprecated. Specify the `--v1` flag to continue using the v1 plugin, or migrate to the supported v2 plugin and use the `--v2` flag. Transition to the oc-mirror v2 plugin for continued support and improvements.

Reviewer: I would add a "for more information..." sentence here just to be safe:

```suggestion
The oc-mirror v1 plugin is deprecated. Specify the `--v1` flag to continue using the v1 plugin, or migrate to the supported v2 plugin and use the `--v2` flag. Transition to the oc-mirror v2 plugin for continued support and improvements.

For more information, see "Mirroring images for a disconnected installation using oc-mirror plugin v2".
```

And at the additional resources section at the bottom of the assembly, add `* xref:../disconnected/about-installing-oc-mirror-v2.adoc#about-installing-oc-mirror-v2[Mirroring images for a disconnected installation using oc-mirror plugin v2]`

toc::[]

[role="_abstract"]

> Running your cluster in a restricted network without direct internet connectivity is possible by installing the cluster from a mirrored set of {product-title} container images in a private registry. This registry must be running whenever the cluster is running. For more information, see Prerequisites.

Reviewer 2: ```suggestion
Running your cluster in a restricted network without direct internet connectivity is possible by installing the cluster from a mirrored set of {product-title} container images in a private registry. This registry must be running whenever the cluster is running. For more information, see the Prerequisites.
```

> Running your cluster in a restricted network without direct internet connectivity is possible by installing the cluster from a mirrored set of {product-title} container images in a private registry. This registry must be running whenever the cluster is running. For more information, see the Prerequisites.

Reviewer 3: more than 300 characters. 

```suggestion
Running your cluster in a restricted network without direct internet connectivity is possible by installing the cluster from a mirrored set of {product-title} container images in a private registry. This registry must be running whenever the cluster is running. For more information, see "Prerequisites".
```

> You can use the oc-mirror OpenShift CLI (`oc`) plugin to mirror images to a mirror registry in your fully or partially disconnected environments. You must run oc-mirror from a system with internet connectivity in order to download the required images from the official Red Hat registries.

Reviewer 2: ```suggestion
You can use the oc-mirror {oc-first} plugin to mirror images to a mirror registry in your fully or partially disconnected environments. You must run oc-mirror from a system with internet connectivity in order to download the required images from the official Red Hat registries.
```

[IMPORTANT]
====
The oc-mirror v1 plugin is deprecated. Specify the `--v1` flag to continue using the v1 plugin, or migrate to the supported v2 plugin and use the `--v2` flag. Transition to the oc-mirror v2 plugin for continued support and improvements.
====

> [role="_additional-resources"]

Reviewer 2: ```suggestion
```

Reviewer 3: not resolved

> .Additional resources

Reviewer 2: ```suggestion
```

Reviewer 3: not resolved

> * xref:../disconnected/installing-mirroring-disconnected.adoc#prerequisites_installing-mirroring-disconnected[Prerequisites]

Reviewer 2: ```suggestion
```
you cannot xref within the same assembly; jump links are not supported; see https://github.com/openshift/openshift-docs/pull/115599/changes#diff-745dd5fa99e444415b1fb594966655b402de789d57275920d596cc3fd3a9fdddL2 for this assembly's ID

Reviewer 3: not resolved

[role="_additional-resources"]
.Additional resources

* xref:../installing/validation_and_troubleshooting/validating-an-installation.adoc#viewing-the-image-pull-source_validating-an-installation[Viewing the image pull source]

> * xref:../disconnected/about-installing-oc-mirror-v2.adoc#installation-oc-mirror-v2-about_about-installing-oc-mirror-v2[About the oc-mirror v2 plugin]

Reviewer: Oh this might be where you moved the link from the deprecation notice. This is fine to keep here if you want, but I would just also add a link to the assembly-level additional resources section.

[role="_additional-resources"]
.Additional resources

* xref:../disconnected/installing-mirroring-disconnected.adoc#oc-mirror-imageset-config-params_installing-mirroring-disconnected[Image set configuration parameters]

> * xref:../disconnected/installing-mirroring-disconnected.adoc#oc-mirror-image-set-examples_installing-mirroring-disconnected[Image set configuration examples]

Reviewer: Was there a specific reason to delete these? They're mentioned in the procedure module, so if not then I would revert this change.

// Mirroring an image set to a mirror registry
include::modules/installing-mirroring-disconnected-mirroring-image-set.adoc[leveloffset=+1]

// Mirroring an image set in a partially disconnected environment

> include::modules/installing-mirroring-disconnected-partial.adoc[leveloffset=+1]

Reviewer: In my opinion, let's just keep the existing docs structure here, which means retaining the existing heading levels as before:

```suggestion
include::modules/installing-mirroring-disconnected-partial.adoc[leveloffset=+2]
```

We could debate whether it should be fixed here or not, but I think we should just keep things as they are as much as possible and just move on

// Mirroring from mirror to mirror

> include::modules/oc-mirror-mirror-to-mirror.adoc[leveloffset=+2]

Reviewer: ```suggestion
include::modules/oc-mirror-mirror-to-mirror.adoc[leveloffset=+3]
```

// Mirroring an image set in a fully disconnected environment

> include::modules/installing-mirroring-disconnected-full.adoc[leveloffset=+1]

Reviewer: ```suggestion
include::modules/installing-mirroring-disconnected-full.adoc[leveloffset=+2]
```

// Mirroring from mirror to disk

> include::modules/oc-mirror-mirror-to-disk.adoc[leveloffset=+2]

Reviewer: ```suggestion
include::modules/oc-mirror-mirror-to-disk.adoc[leveloffset=+3]
```

// Mirroring from disk to mirror in a disconnected environment

> include::modules/oc-mirror-disk-to-mirror.adoc[leveloffset=+2]

Reviewer: ```suggestion
include::modules/oc-mirror-disk-to-mirror.adoc[leveloffset=+3]
```

> * xref:../disconnected/installing-mirroring-disconnected.adoc#oc-mirror-image-set-examples_installing-mirroring-disconnected[Image set configuration examples]

Reviewer 2: ```suggestion
```
no jump links

Reviewer 3: not resolved

* xref:../disconnected/updating/disconnected-update-osus.adoc#updating-disconnected-cluster-osus[Using the OpenShift Update Service in a disconnected environment]

// Mirroring an image set to a mirror registry
include::modules/installing-mirroring-disconnected-mirroring-image-set.adoc[leveloffset=+1]

[role="_additional-resources"]
.Additional resources

> * xref:../disconnected/installing-mirroring-disconnected.adoc#mirroring-image-set-partial_{context}[Mirroring an image set in a partially disconnected environment]

Reviewer 2: ```suggestion
```
no jump links

Reviewer 3: not resolved

> * xref:../disconnected/installing-mirroring-disconnected.adoc#mirroring-image-set-full_{context}[Mirroring an image set in a fully disconnected environment]

Reviewer 2: ```suggestion
```
no jump links

Reviewer 3: not resolved

// Mirroring an image set in a partially disconnected environment
include::modules/installing-mirroring-disconnected-partial.adoc[leveloffset=+1]

// Mirroring from mirror to mirror
include::modules/oc-mirror-mirror-to-mirror.adoc[leveloffset=+2]

// Mirroring an image set in a fully disconnected environment
include::modules/installing-mirroring-disconnected-full.adoc[leveloffset=+1]

// Mirroring from mirror to disk
include::modules/oc-mirror-mirror-to-disk.adoc[leveloffset=+2]

// Mirroring from disk to mirror in a disconnected environment
include::modules/oc-mirror-disk-to-mirror.adoc[leveloffset=+2]

// Configuring your cluster to use the resources generated by oc-mirror
include::modules/oc-mirror-updating-cluster-manifests.adoc[leveloffset=+1]

[role="_additional-resources"]
.Additional resources

* xref:../extensions/catalogs/managing-catalogs.adoc#olmv1-adding-a-catalog-to-a-cluster_managing-catalogs[Adding a catalog to a cluster]

> * xref:../disconnected/installing-mirroring-disconnected.adoc#mirroring-image-set-partial_{context}[Mirroring an image set in a partially disconnected environment]

Reviewer 2: ```suggestion
```

Reviewer 3: not resolved

> * xref:../disconnected/installing-mirroring-disconnected.adoc#mirroring-image-set-full_{context}[Mirroring an image set in a fully disconnected environment]

Reviewer 2: ```suggestion
```

Reviewer 3: not resolved

> * xref:../disconnected/installing-mirroring-disconnected.adoc#oc-mirror-mirror-to-disk_installing-mirroring-disconnected[Mirroring from mirror to disk]

Reviewer 2: ```suggestion
```

Reviewer 3: not resolved

> * xref:../disconnected/installing-mirroring-disconnected.adoc#oc-mirror-disk-to-mirror_installing-mirroring-disconnected[Mirroring from disk to mirror in a disconnected environment]

Reviewer 2: ```suggestion
```

Reviewer 3: not resolved

.Additional resources

> * xref:../disconnected/installing-mirroring-disconnected.adoc#oc-mirror-image-set-examples_installing-mirroring-disconnected[Image set configuration examples]

Reviewer 2: ```suggestion
```

Reviewer 3: not resolved

> * xref:../disconnected/installing-mirroring-disconnected.adoc#mirroring-image-set-partial_{context}[Mirroring an image set in a partially disconnected environment]

Reviewer 2: ```suggestion
```
couple of problems here with all of these links:
adding _{context} tags to xrefs is incorrect
see the example xref in  https://docs.google.com/document/d/1sPonVNH_0wvv4Q-Ls00fvJsJsqLRXZkUUjv3xdBOHPg/edit?tab=t.0
also, self-referential links are not allowed; so you cannot have an xref to a module in the local assembly (in other words, no jump links)

Reviewer 3: not resolved

> * xref:../disconnected/installing-mirroring-disconnected.adoc#mirroring-image-set-full_{context}[Mirroring an image set in a fully disconnected environment]

Reviewer 2: ```suggestion
```

Reviewer 3: not resolved

> * xref:../disconnected/installing-mirroring-disconnected.adoc#oc-mirror-updating-cluster-manifests_installing-mirroring-disconnected[Configuring your cluster to use the resources generated by oc-mirror]

Reviewer 2: ```suggestion
```

Reviewer 3: not resolved
